syncstorage/tests/functional/test_storage.py

```diff
@@ -1575,7 +1575,7 @@ def test_batch_with_failing_bsos(self):
         endpoint = collection + '?batch={0}&commit=true'.format(batch)
         resp = self.app.post_json(endpoint, bsos)
         self.assertEqual(len(resp.json['failed']), 1)
-        self.assertEqual(len(resp.json['success']), 1)
+        self.assertEqual(len(resp.json['success']), 2)
```

Comment:
The "a" item was already reported as successful in the previous POST.  If we include it here as well, the semantics of the "success" list change between non-batch uploads (where it's only items from the request body) and batch uploads (where it would be any items from the batch).